### ISSUE TITLE
ResNet18 seems to have more blocks than usual

### ISSUE THREAD

**Issue by robertomest** [Microsoft/CNTK#2873]
Hello,

I was reviewing the ResNet code and pretrained models and it seems that the ResNet18 has one extra block at the last stage (so it is actually a ResNet20). If you check the number of parameters it has:
`Training 16410152 parameters in 68 parameter tensors.`
It is substantially larger than the number I found, which is the same as the pytorch implementation:
`Training 11689512 parameters in 62 parameter tensors.`
I believe the 6 extra parameter tensors are 2 convolution weights, 2 scale and 2 bias parameters for batchnorm, that belong to that extra block. On the ResNet training code, the number of blocks for ResNet18 is `[2, 1, 1, 2]`, instead of `[2, 1, 1, 1]`. ResNet34, on the other hand, seems to have the usual number of blocks.

**Comment by cha-zhang** [comment]
Indeed the ResNet18 model should be [2,1,1,1]. This should be fixed. Would you please send in a pull request?

**Comment by robertomest** [comment]
Yes, of course. Should the current implementation be erased or changed to ResNet20? Although there is no "official" ResNet20, this network seems to work just fine and the weights are already available.

**Comment by cha-zhang** [comment]
yeah, we can call the old one ResNet20. :)
